lib/tlscontext.c

```diff
@@ -424,9 +424,6 @@ tls_context_setup_verify_mode(TLSContext *self)
 
   switch (self->verify_mode)
     {
-    case TVM_NONE:
```

Comment:
I've checked if there was any reason in the past to handle "none" and "optional-trusted" in separate cases, but I haven't found anything.
The original [commit](https://github.com/syslog-ng/syslog-ng/commit/0a287bd61240503a970e6431d4ebfb65d42b469d) set `SSL_VERIFY_NONE` in both cases, so we should merge the 2 options into the same case.